### ISSUE TITLE
Few suggestions for module configuration

### ISSUE THREAD

**Issue by gbrandt1** [VCVRack/Rack#683]
the main .cpp file for module libraries doubles much of the information of the json manifest.
instead of hardcoding the slug etc. and addModules, the configuration could be read from the
manifest at runtime.

the manifest should then also contain information about all the modules.
this would be useful also to know more about the modules before running / instantiating.
for example to build lists, counts or .vcv files for one manufacturer (generate screenshots eg.)

one important info that should be added to the module tags is the minimum HP a module uses

**Comment by AndrewBelt** [comment]
Thanks for reminding me. I've deprecated `website` and `manual` in 663134c.

**Comment by WIZARDISHUNGRY** [comment]
@gbrandt1 Are there modules with dynamic widths?

**Comment by AndrewBelt** [comment]
@WIZARDISHUNGRY Yes, Core Blank is one. HP is a pretty useless specification with VCV Rack. Normally you only care in hardware when you have limited rack space.

**Comment by gbrandt1** [comment]
i have limited screen space. and even if that is no consideration people sometimes like to fill gaps or constrain the HP used. there's a reason entire series or even companies are formed around number of HP and it's a filter option on modulargrid.
also as i mention above it could also be useful for script based creation of vcv rack patches because there the position needs to be known.

in response to Jon i believe the most popular resizable module is JW full scope

**Comment by AndrewBelt** [comment]
Well, I can't argue with that if you treat Rack to be as close as possible to a real modular. That's what I designed Rack to be...
